src/mantarray_desktop_app/mc_comm.py

```diff
@@ -30,21 +80,117 @@ def __init__(self, *args: Any, **kwargs: Any):
             self._board_queues
         )
 
+    def _setup_before_loop(self) -> None:
+        msg = {
+            "communication_type": "log",
+            "message": f'Microcontroller Communication Process initiated at {datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S.%f")}',
+        }
+        to_main_queue = self._board_queues[0][1]
+        if not self._suppress_setup_communication_to_main:
+            to_main_queue.put(msg)
+        self.create_connections_to_all_available_boards()
+
+        board_idx = 0
+        board = self._board_connections[board_idx]
+        if isinstance(
+            board, MantarrayMcSimulator
+        ):  # pragma: no cover  # Tanner (3/16/21): it's impossible to connect to any serial port in CI, so _setup_before_loop will always be called with a simulator and this if statement will always be true in pytest
```

Comment:
potential future consideration: subclass the pyserial interface so that it has a dummy `start` method and `is_start_up_complete` event (and the `start` sets the event)

maybe overkill, maybe helpful, not sure